protocol/src/main/java/org/threadly/litesockets/protocols/http/request/HTTPRequestHeader.java

```diff
@@ -1,38 +1,41 @@
-package org.threadly.litesockets.protocol.http.structures;
+package org.threadly.litesockets.protocols.http.request;
 
+import java.nio.ByteBuffer;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
-import org.threadly.litesockets.protocol.http.structures.HTTPConstants.REQUEST_TYPE;
+import org.threadly.litesockets.protocols.http.shared.HTTPConstants;
+import org.threadly.litesockets.protocols.http.shared.HTTPUtils;
+import org.threadly.litesockets.protocols.http.shared.RequestType;
 
 
 /**
  * This is an immutable HTTP Request Header.  Basically the first line of the http request.  
  */
 public class HTTPRequestHeader {
-  public static final int REQUIRED_REQUEST_ITEMS = 3;
-  public final String rawRequest;
-  public final String requestType;
-  public final String requestPath;
-  public final Map<String, String> requestQuery;
-  public final String httpVersion;
+  private static final int REQUIRED_REQUEST_ITEMS = 3;
+  private final String rawRequest;
+  private final String requestType;
+  private final String requestPath;
+  private final Map<String, String> requestQuery;
+  private final String httpVersion;
   
-  public HTTPRequestHeader(String rawRequest) {
-    this.rawRequest = rawRequest.intern();
-    String[] tmp = rawRequest.split(" ");
+  public HTTPRequestHeader(final String requestHeader) {
+    this.rawRequest = requestHeader.trim().intern();
```

Comment:
I don't think this needs to be changed, but be aware how the intern behavior has changed dramatically between java versions:
http://java-performance.info/string-intern-in-java-6-7-8/